mrp_production_split/wizards/mrp_production_split_wizard.py

```diff
@@ -0,0 +1,167 @@
+        lang = self.env["res.lang"]._lang_get(self.env.user.lang)
+        try:
+            return float(
+                value.replace(lang.thousands_sep, "")
+                .replace(lang.decimal_point, ".")
+                .strip()
+            )
+        except ValueError as e:  # pragma: no cover
+            raise UserError(_("%s is not a number.", value)) from e
+
+    @api.model
+    def _parse_float_list(self, value: str) -> List[float]:
+        """Parse a list of float numbers from a string."""
+        return [self._parse_float(v) for v in value.split()]
+
+    @api.onchange("custom_quantities")
+    def _onchange_custom_quantities_check(self):
+        """Check that the custom quantities are valid."""
+        if self.custom_quantities:  # pragma: no cover
+            try:
+                self._parse_float_list(self.custom_quantities)
+            except UserError:
+                return {
+                    "warning": {
+                        "title": _("Invalid quantities"),
+                        "message": _("Please enter a space separated list of numbers."),
+                    }
+                }
+
+    def _get_split_quantities(self) -> List[float]:
+        """Return the quantities to split, according to the settings."""
+        production = self.production_id
+        rounding = production.product_uom_id.rounding
+        if self.split_mode == "simple":
+            if (
+                fields.Float.compare(
+                    self.split_qty, production._get_quantity_to_backorder(), rounding
+                )
+                > 0
+            ):
+                raise UserError(_("You can't split quantities already in production."))
+            if fields.Float.is_zero(self.split_qty, precision_rounding=rounding):
+                raise UserError(_("Nothing to split."))
+            return [production.product_qty - self.split_qty, self.split_qty]
+        elif self.split_mode == "equal":
+            split_total = min(production._get_quantity_to_backorder(), self.split_qty)
+            split_count = int(split_total // self.split_equal_qty)
+            split_rest = production.product_qty - split_total
+            split_rest += split_total % self.split_equal_qty
+            quantities = [self.split_equal_qty] * split_count
+            if not fields.Float.is_zero(split_rest, precision_rounding=rounding):
+                quantities = [split_rest] + quantities
+            return quantities
+        elif self.split_mode == "custom":
+            quantities = self._parse_float_list(self.custom_quantities)
+            split_total = sum(quantities)
+            split_rest = production.product_qty - split_total
+            if not fields.Float.is_zero(split_rest, precision_rounding=rounding):
+                quantities = [split_rest] + quantities
+            return quantities
+        else:  # pragma: no cover
+            raise UserError(_("Invalid Split Mode: '%s'", self.split_mode))
```

Comment:
```suggestion
            raise UserError(_("Invalid Split Mode: '%(mode)s'" % {"mode": self.split_mode}))
```

Reply:
I do tend to do this when `len(args) > 1`, definitely a must for translations, but not if `len(args) == 1` as it seems redundant.

Reply:
fact is: in v16 you'll have to change it ;)

Reply:
😱 what am I missing?